### DISCUSSION TITLE
Is there a way to create a visualisation based on a specific task inside a file ?

### DISCUSSION THREAD

**Discussion by mathisgauthey** [Richardsl/heatmap-calendar-obsidian#88] [Q&A]
Hi everyone. As stated in my [obsidian workflow guide](https://mathisgauthey.github.io/my-complete-obsidian-workflow-to-manage-my-life/), or inside my [example vault](https://github.com/mathisgauthey/obsidian-workflow-template), I'm using recurring tasks query from the Tasks Plugin to show the recurring tasks inside my daily notes.

The real recurring tasks are inside a project file, along with each completed occurence.

Is there a way to use this project file with the occurence using the standard Task's format `✅ YYYY-MM-DD` to create a heatmap visualisation of my current recurring tasks ? 

Something along the line of :
1. Search for recurring tasks names due today.
2. Find the file for each recurring tasks and query their completed date.
4. Create a heatmap calendar per current habit automatically based on previous queries.

If that looks doable, maybe someone did it already and would be willing to share their work. Or maybe I could try finding it myself but here's a discussion to come up with something!

**Comment by mmorowitz** [comment]
I'd like to do this as well. I wonder if it's just more of a Dataview query problem than an issue with this plugin. 

Is there a way in Dataview to query the property and then base the boolean off of the checkbox on the same line? Something like this? https://forum.obsidian.md/t/using-dataviewjs-to-query-keywords-in-nested-tasks-checkboxes/42876

I haven't tried to implement that yet, but that's basically what I'm looking for.

**Reply by mathisgauthey** [reply]
Hey there, I also asked in the obsidian-task plugin discussion if you want to check the [answers](https://github.com/obsidian-tasks-group/obsidian-tasks/discussions/2205) !
